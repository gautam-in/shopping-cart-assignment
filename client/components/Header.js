import Link from 'next/link';
import styled from 'styled-components';
import Nav from './Nav';

const Logo = styled.img`
  font-size: 4rem;
  margin-left: 2rem;
  position: relative;
  z-index: 2;
  a,img {
    color: white;
    text-decoration: none;
    text-transform: uppercase;
    padding: 0.5rem 1rem;
    width: 100px;
    height: 100px;
  }
`;

const HeaderStyles = styled.header`
  .bar {
    display: flex;
    flex-direction: row;
  }
  .sub-bar {
    display: grid;
    grid-template-columns: 1fr auto;
    border-bottom: 1px solid var(--secondary, black);
  }

  .separator {
    height: 10px;
    width: 100%;
  }
`;

export default function Header() {
  return (
    <HeaderStyles>
      <div className="bar">
         <img loading="lazy" src="/static/images/logo.png" loading="lazy" alt="Sabka Bazar" width="100px" height="50px"/>
         <Nav />
      </div>
      <div className="gradient separator"></div>
      <div className="sub-bar">

      </div>
    </HeaderStyles>
  );
}