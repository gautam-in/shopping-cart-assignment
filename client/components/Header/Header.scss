$light-gray:#c8cbce61;

.header {
  background: white;
  box-shadow: 0px 1px 10px rgba(0, 0, 0, 0.3);
  height: 6rem;
  width: 100%;
  position: fixed;
  top: 0;
  z-index: 1;

  .navbar {
    height: 100%;
    display: flex;
    align-items: center;

    a {
      margin: 0;
      padding: 10px;
      color: gray; //
      text-align: center;
      text-decoration: none;
      display: inline-block; //
      font-size: 0.8em;
      font-weight: bold;

      &:hover {
        color: black;
      }
    }

    .brand-name {
      .brand-logo {
        // height: 4rem;
        width: auto;
      }
    }

    .left-nav {
      flex-grow: 4;

      .tabs {
        display: flex;
        margin: 2em 0 0;

        li:hover {
          background: $light-gray;
        }
      }
    }

    .right-nav {
      display: flex;
      flex-direction: column;
      justify-content: space-between;
      height: 100%;

      .links {
        display: flex;
        justify-content: flex-end;
        margin: 0;
        padding: 5px 10px 10px 0px;

        li {
          cursor: pointer;

          &:hover {
            background: $light-gray;
          }
        }
      }

      .my-cart {
        a {
          display: flex;
          align-items: center;
          background-color: $light-gray;

          .cart-icon {
            padding: 0 1em;
          }
        }

        span {
          padding-right: 25px;
        }

        display: flex;
        justify-content: flex-end;
      }
    }

    .tabs,
    .links {
      list-style-type: none;
    }
  }
}

//ipad & above
@media only screen and(min-width:1030px) {
  .header {
    .navbar {
      padding: 0 6rem;
    }
  }
}

// mobile
@media only screen and(max-width:420px) {
  .header {
    height: 4rem;

    .navbar {
      .brand-name {
        .brand-logo {
          height: 3rem;
        }
      }

      .left-nav {
        .tabs {
          display: none;
        }
      }

      .right-nav {
        .links {
          display: none;
        }

        .my-cart {
          flex-grow: 1;
        }
      }
    }
  }


}