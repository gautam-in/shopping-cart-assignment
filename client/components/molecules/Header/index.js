import Image from "next/image";
import Link from "next/link";
import { useRouter } from "next/router";
import {
  HeaderWrapper,
  Navigation,
  CartContainer,
  CartButton,
  CartIcon,
  ItemsCount,
  HeaderContainer,
} from "./Header.styles";
import Logo from "../../../public/images/logo_2x.png";
import { useCart } from "../../../global/utils/useCart";
import CartSection from "../CartSection";
import { useCallback } from "react";

const Header = () => {
  const { toggleCart, getCartItemsCount } = useCart();
  const router = useRouter();
  const cartItemsCount = getCartItemsCount();
  const cartButtonLabel = `item${cartItemsCount !== 1 ? "s" : ""}`;

  const onLogoClick = useCallback(() => {
    router.push("/");
  });

  return (
    <HeaderWrapper>
      <HeaderContainer>
        <Image
          src={Logo}
          alt="Sabka Bazaar logo"
          width="142"
          height="64"
          onClick={onLogoClick}
        />
        <Navigation aria-label="Site navigation">
          <Link href="/home">Home</Link>
          <Link href="/products">Products</Link>
        </Navigation>
        <CartContainer>
          <Navigation aria-label="login navigation">
            <Link href="/login">Sign in</Link>
            <Link href="/register">Register</Link>
          </Navigation>
          <CartButton
            aria-label="Cart items"
            title={`Cart button ${cartItemsCount} ${cartButtonLabel}`}
            onClick={toggleCart}
          >
            <CartIcon />
            <ItemsCount>
              {cartItemsCount} <span>{cartButtonLabel}</span>
            </ItemsCount>
          </CartButton>
        </CartContainer>
      </HeaderContainer>
      <CartSection />
    </HeaderWrapper>
  );
};

export default Header;
