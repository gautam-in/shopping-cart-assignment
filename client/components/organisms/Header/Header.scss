@import "../../../styles/variables";
@import "../../../styles/mq-mixin";

header {
  display: flex;
  /* box-shadow: 0px 11px 9px -8px $bg-secondary-color; */
  height: $mobile-header-height;

  $entry_access_height: 30px;
  .entry_access {
    height: $entry_access_height;
    text-align: right;
    font-size: $primary-font-size;
    font-weight: 600;
  }

  .header_content_alignment {
    display: flex;
    justify-content: flex-end;
    height: 100%;
  }

  @include mq(tab) {
    height: $tab-header-height;
    .header_content_alignment {
      justify-content: space-between;
      height: calc(100% - #{$entry_access_height});
    }
  }
}
