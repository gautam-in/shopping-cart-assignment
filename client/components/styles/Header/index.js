import styled  from 'styled-components';

export const Header = styled.header`
    box-shadow: 0 0 12px rgba(0,0,0,0.3);  
`
export const HeaderContainer = styled.div`
    width : 85%;
    margin: 0 auto;
    display: flex;
    gap: 1.2rem;
    align-items: center;
    padding: 1.2rem 0;

    img {
        height: 4.2rem;
        margin-right: 2.4rem;
    }

    .btn-mobile-icon {
        border: none;
        background: none;
        cursor: pointer;
        display: none;

        @media (max-width: 38em) {
            display: block;
        }   
    }

    .headerMobileView {
        position: absolute;
        top: 0;
        left: 0;
        background-color: rgba(255, 255, 255, 0.9);
        display: flex;
        width: 100%;
        height: 100vh;
        color: black;
        font-size: 2.4rem;
        font-weight: bold;
        line-height: 2.5;
        flex-direction: column;
        align-items: center;
        justify-content: center;
        display: none;
    }
`
export const HeaderContent = styled.div`
    width: 100%;

    nav {
        display: flex;
        align-items: center;
        gap:1.2rem;  
        
        @media (max-width: 38em) {
            display: none;
        }   
    }

    .topNav,.bottomNav {
        justify-content: end;
    }

    .bottomNav {
        display: flex;
        align-items: center;
        margin-top: 0.8rem;
        gap : 1.2rem;
        justify-content: space-between;
        position: relative;
    }

    nav a:link, nav a:visited {
        display: inline-block;
        text-decoration: none;
        color: #333;
        font-weight: 600;
        font-size: 1.4rem;
        transition: all 0.3s;
        border: none;
    }

    nav a:hover,
    nav a:active {
        color: #cf711f;
    }

`

export const CartButton = styled.a`
        display: inline-block;
        text-decoration: none;
        color: #fff;
        font-weight: 500;
        font-size: 1.4rem;
        border: none;
        padding: 0.8rem 1.2rem;  
        cursor: pointer;
        border-radius: 5px;
        background-color: grey;

        @media (max-width: 38em) {
            position: absolute;
            right: 0;
        }   
`






