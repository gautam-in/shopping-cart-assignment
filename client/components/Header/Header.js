import React, { useContext } from "react";
import "./Header.scss";
import { Link } from "react-router-dom";
import { useSelector } from "react-redux";
import AuthContext from "../../AuthContext";

const Header = () => {
  const { userAuthentication, toggleUserAuthentication } = useContext(
    AuthContext
  );
  const cartItemsCount = useSelector(
    (state) =>
      state.cart && state.cart.itemsAdded && state.cart.itemsAdded.length
  );
  return (
    <header className="header">
      <nav className="navbar">
        <div className="brand-name">
          <img
            className="brand-logo"
            src="../static/images/logo.png"
            alt="Sabka Bazaar Logo"
            height="60"
            width="100"
          />
        </div>
        <div className="left-nav">
          <ul className="tabs">
            <li>
              <Link to="/">Home</Link>
            </li>
            <li>
              <Link to="/products">Products</Link>
            </li>
          </ul>
        </div>
        <div className="right-nav">
          {userAuthentication === "logged-in" ? (
            <ul className="links">
              <li
                onClick={() => {
                  sessionStorage.setItem("status", "");
                  toggleUserAuthentication();
                }}
              >
                <a href="#">Logout</a>
              </li>
            </ul>
          ) : (
            <ul className="links">
              <li>
                <Link to="/login">SignIn</Link>
              </li>
              <li>
                <Link to="/sign-up">Register</Link>
              </li>
            </ul>
          )}
          <div className="my-cart">
            <Link to="/cart">
              <img
                className="cart-icon"
                src="../static/images/cart.svg"
                alt="cart image"
                height="27"
                width="30"
              />
              <span
                style={
                  cartItemsCount > 0 ? { color: "#ea1d71" } : { color: "gray" }
                }
              >
                {cartItemsCount} {cartItemsCount === 1 ? "item" : "items"}
              </span>
            </Link>
          </div>
        </div>
      </nav>
    </header>
  );
};

export default Header;
