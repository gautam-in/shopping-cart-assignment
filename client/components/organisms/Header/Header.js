import React, { useContext, useState } from "react";
import HeaderLogo from "../../atoms/HeaderLogo/HeaderLogo";
import NavigationLinks from "../../molecules/NavigationLinks/NavigationLinks";
import CartButton from "../../molecules/CartButton/CartButton";
import EntryAccessButton from "../../atoms/EntryAccessButton";
import Modal from "../../molecules/Modal/Modal";

import "./Header.scss";
import { useSelector } from "react-redux";
import { DeviceContext } from "../../../App";
import Cart from "../../molecules/Cart/Cart";
import { useHistory } from "react-router";

function Header() {
  const { isMobile, isDesktop } = useContext(DeviceContext);
  const history = useHistory();
  const itemsCount = useSelector((state) => state.addItems.itemsCount);

  const [showModal, setShowModal] = useState(false);

  const onCardClick = () => {
    if (isDesktop) setShowModal(true);
    else history.push("/cart");
  };

  return (
    <header className="parent_spacing">
      <HeaderLogo />
      <div className="width_full">
        {!isMobile && (
          <div className="entry_access">
            <EntryAccessButton linkTo="/login">SignIn</EntryAccessButton>
            <EntryAccessButton linkTo="/register">Register</EntryAccessButton>
          </div>
        )}

        <div className="header_content_alignment">
          {!isMobile && <NavigationLinks />}
          <CartButton handleOpen={onCardClick} />
          {/* {showModal && ( */}
          <Modal
            isOpen={showModal}
            handleClose={() => setShowModal(false)}
            title={
              itemsCount === 0 ? `My Cart` : `My Cart (${itemsCount} items)`
            }
          >
            <Cart handleClose={() => setShowModal(false)} />
          </Modal>
          {/* )} */}
        </div>
      </div>
    </header>
  );
}

export default Header;
