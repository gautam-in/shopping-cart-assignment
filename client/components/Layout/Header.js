import React from 'react'
import Nav from './Nav'
import styles from '../../styles/header.module.scss'
import Link from 'next/link'


const Header = () => {
    return (
        <header className={styles.headerContainer}>
            <div className={styles.header}>
                <div className={styles.logo}>
                    <Link href="/">
                        <img src="/static/images/logo.png" alt='img' />
                    </Link>
                </div>
                <Nav />
            </div>
        </header>
    )
}

export default Header