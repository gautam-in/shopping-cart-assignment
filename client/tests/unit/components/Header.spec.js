import { mount, createLocalVue, RouterLinkStub } from "@vue/test-utils";
import Header from "@/components/Header.vue";
import VueRouter from "vue-router";
import Vuex from "vuex";
import Vue from "vue";
Vue.use(Vuex);
const localVue = createLocalVue();
localVue.use({
  Header,
  VueRouter,
  Vuex,
});

describe("Header", () => {
  let actions;
  let $store;
  let state;

  const push = jest.fn();
  const $router = {
    push: jest.fn(),
  };

  beforeEach(() => {
    // actions = {
    //   actionClick: jest.fn(),
    //   actionInput: jest.fn(),
    // };
    (state = {
      cartItems: [],
      loggedIn: false,
    }),
      ($store = new Vuex.Store({
        state,
        actions,
      }));
  });

  it("snapshot", () => {
    const wrapper = mount(Header, {
      localVue,
      propsData: {
        cartItemsData: [],
      },
      mocks: {
        $store: {
          state: {
            cartItems: [],
            loggedIn: false,
          },
        },
        $router,
      },
      stubs: {
        RouterLink: RouterLinkStub,
      },
    });
    expect(wrapper).toMatchSnapshot();
    // expect(wrapper.contains("div")).toBe(true);
  });
});
