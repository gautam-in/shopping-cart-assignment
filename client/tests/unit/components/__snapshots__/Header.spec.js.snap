// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header snapshot 1`] = `
<section class="container" cartitemsdata="">
  <div class="row">
    <div class="col-6 col-sm-6 col-md-4 col-lg-4 col-xl-3"><img id="logo" src="../assets/images/logo.png" alt="logo"></div>
    <div id="nav" class="col-6 col-sm-6 col-md-4 col-lg-4 col-xl-5"><a>Home</a> <a>Products</a></div>
    <div id="cartContainer" class="col-6 col-sm-6 col-md-4 col-lg-4 col-xl-4">
      <div class="linkContainer"><a>Sign In</a> <a>Register</a></div>
      <div class="cartStyle"><img src="../assets/images/cart.svg" alt="cart">
        0 items
        <div class="badgeStyle">0</div>
      </div>
    </div>
  </div>
  <!---->
  <!---->
</section>
`;
